### PR TITLE
[Feature] Support YOLOv5 instance segmentation

### PR DESCRIPTION
**Pull request by JosonChan1998**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support YOLOv5 instance segmentation

The reproduced mAP is listed below:
YOLOv5-s| box AP | mask AP |
--- |:---:|:---:|
origin   | 37.7 | 31.7 | 
ours| 38.0 | 32.1 |

### CODE REVIEW

File: configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        head_module=dict(
+            type='YOLOv5InsHeadModule', mask_channels=32, proto_channels=256),
+        mask_overlap=mask_overlap,
+        loss_mask=dict(
+            type='mmdet.CrossEntropyLoss', use_sigmoid=True, reduction='none'),
+        loss_mask_weight=0.05),
+    test_cfg=model_test_cfg)
+
+pre_transform = [
+    dict(type='LoadImageFromFile'),
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py:59]: dict(type='LoadImageFromFile', backend_args=_base_.backend_args),

File: configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        type='YOLOv5RandomAffine',
+        max_rotate_degree=0.0,
+        max_shear_degree=0.0,
+        scaling_ratio_range=(1 - _base_.affine_scale, 1 + _base_.affine_scale),
+        border=(-_base_.img_scale[0] // 2, -_base_.img_scale[1] // 2),
+        border_val=(114, 114, 114),
+        min_area_ratio=min_area_ratio,
+        max_aspect_ratio=max_aspect_ratio,
+        use_mask_refine=use_mask2refine),
+    dict(
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py:84]: 加一个TODO: 说明暂时没有考虑mask，因此 albu里面不能有几何变换

File: configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        mask_overlap=mask_overlap,
+        coco_style=False),
+    dict(
+        type='PackDetInputs',
+        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape', 'flip',
+                   'flip_direction'))
+]
+
+test_pipeline = [
+    dict(type='LoadImageFromFile'),
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py:109]: dict(type='LoadImageFromFile', file_client_args=_base_.file_client_args),

File: configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    dict(type='YOLOv5KeepRatioResize', scale=_base_.img_scale),
+    dict(
+        type='LetterResize',
+        scale=_base_.img_scale,
+        allow_scale_up=False,
+        half_pad_param=half_pad_param,
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py:115]: 需要解释下这个参数

File: configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        type='mmdet.PackDetInputs',
+        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
+                   'scale_factor', 'pad_param'))
+]
+
+train_dataloader = dict(dataset=dict(pipeline=train_pipeline))
+val_dataloader = dict(
+    batch_size=val_batch_size_per_gpu,
+    num_workers=val_num_workers,
+    dataset=dict(pipeline=test_pipeline, batch_shapes_cfg=batch_shapes_cfg))
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py:128]:
```suggestion
    dataset=dict(pipeline=test_pipeline))
```

File: mmyolo/datasets/transforms/formatting.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                        results[key][valid_idx])
+                    ignore_instance_data[self.mapping_table[key]] = to_tensor(
+                        results[key][ignore_idx])
+                else:
+                    instance_data[self.mapping_table[key]] = to_tensor(
+                        results[key])
+        data_sample.gt_instances = instance_data
+        data_sample.ignored_instances = ignore_instance_data
+
+        if 'proposals' in results:
```
Comment (hhaAndroid) [inline mmyolo/datasets/transforms/formatting.py:81]: 我们不会支持 proposals，可以删掉这个代码

File: mmyolo/datasets/transforms/formatting.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+                scores=to_tensor(results['proposals_scores']))
+            data_sample.proposals = proposals
+
+        if 'gt_seg_map' in results:
+            gt_sem_seg_data = dict(
+                sem_seg=to_tensor(results['gt_seg_map'][None, ...].copy()))
+            data_sample.gt_sem_seg = PixelData(**gt_sem_seg_data)
+
+        # Only modified here
+        if 'gt_panoptic_seg' in results:
```
Comment (hhaAndroid) [inline mmyolo/datasets/transforms/formatting.py:93]: 麻烦多写一点注释： 为了将支持实例分割标注 overlap 即 (h,w) 格式的实例分割 标注，我们使用了  gt_panoptic_seg 字段来统一建模

File: mmyolo/datasets/transforms/transforms.py [hunk 8]
```diff
@@ -99,17 +99,17 @@ def _resize_img(self, results: dict):
                                             self.scale)
 
             if ratio != 1:
-                # resize image according to the ratio
-                image = mmcv.imrescale(
+                # resize image according to the shape
+                image = mmcv.imresize(
```
Comment (hhaAndroid) [inline mmyolo/datasets/transforms/transforms.py:103]: 加一个注意事项说明：我们目前在 COCO 上面测试修改这段代码不会对检测结果照成影响。如果你发现对你结果有影响，可以联系我们

File: mmyolo/datasets/transforms/transforms.py [hunk 9]
```diff
@@ -375,8 +391,10 @@ class LoadAnnotations(MMDET_LoadAnnotations):
     def __init__(self,
                  mask2bbox: bool = False,
                  poly2mask: bool = False,
-                 **kwargs) -> None:
+                 merge_polygons: bool = True,
```
Comment (hhaAndroid) [inline mmyolo/datasets/transforms/transforms.py:394]: 加一下 docstring

Comment (hhaAndroid) [reply mmyolo/datasets/transforms/transforms.py:394]: 以及说明下为啥要合并： 存储结构更加简单，训练更加高效，特别是在某个 bbox 内部的 mask 被分成多个 polygon 的时候

Comment (hhaAndroid) [reply mmyolo/datasets/transforms/transforms.py:394]: 如果不合并，会影响精度吗？

File: mmyolo/datasets/transforms/transforms.py [hunk 10]
```diff
@@ -503,6 +523,72 @@ def _load_masks(self, results: dict) -> None:
         gt_masks = PolygonMasks([mask for mask in gt_masks], h, w)
         results['gt_masks'] = gt_masks
 
+    def merge_multi_segment(self,
```
Comment (hhaAndroid) [inline mmyolo/datasets/transforms/transforms.py:526]: 这段代码可以稍微优化下，让他更易读

File: mmyolo/datasets/transforms/transforms.py [hunk 11]
```diff
@@ -1555,3 +1708,127 @@ def transform(self, results: dict) -> dict:
...
+            the index of gt masks. If set to False, one mask is a binary mask.
+            Default to True.
+        coco_style (bool): Whether to use coco_style to convert the polygons
+            to bitmaps.
+    """
+
+    def __init__(self,
+                 downsample_ratio: int = 4,
+                 mask_overlap: bool = True,
+                 coco_style: bool = False):
```
Comment (hhaAndroid) [inline mmyolo/datasets/transforms/transforms.py:1732]: 加一下说明：这个代码只是用于测试训练速度是否有提升，我们不建议你设置为 True

File: mmyolo/datasets/transforms/transforms.py [hunk 12]
```diff
@@ -1555,3 +1708,127 @@ def transform(self, results: dict) -> dict:
...
+            # NOTE: fillPoly firstly then resize is trying the keep the same
+            #  way of loss calculation when mask-ratio=1.
+            mask = mmcv.imresize(mask, (nw, nh))
+        return mask
+
+    def polygons2masks(self,
+                       img_shape: Tuple[int, int],
+                       polygons: PolygonMasks,
+                       color: int = 1) -> np.ndarray:
+        """Return (640, 640) non-overlap mask."""
```
Comment (hhaAndroid) [inline mmyolo/datasets/transforms/transforms.py:1778]: 麻烦简单完善下 docstr 和代码

File: mmyolo/models/dense_heads/yolov5_head.py [hunk 13]
```diff
@@ -95,7 +95,8 @@ def init_weights(self):
             b = mi.bias.data.view(self.num_base_priors, -1)
             # obj (8 objects per 640 image)
             b.data[:, 4] += math.log(8 / (640 / s)**2)
-            b.data[:, 5:] += math.log(0.6 / (self.num_classes - 0.999999))
+            b.data[:, 5:5 + self.num_classes] += math.log(
```
Comment (hhaAndroid) [inline mmyolo/models/dense_heads/yolov5_head.py:98]: 如果这个地方很关键，可以加一下详细说明

File: configs/yolov5/ins_seg/yolov5_ins_m-v61_sync_bn_fast_8xb16-300e_coco_instance.py [hunk 14]
```diff
@@ -0,0 +1,89 @@
+_base_ = './yolov5_ins_s-v61_sync_bn_fast_8xb16-300e_coco_instance.py'
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_m-v61_sync_bn_fast_8xb16-300e_coco_instance.py:1]: configs/yolov5/ins_seg/yolov5_ins_m-v61_sync_bn_fast_8xb16-300e_coco_instance.py
-> 
configs/yolov5/ins_seg/yolov5_ins_m-v61_syncbn_fast_8xb16-300e_coco_instance.py

File: configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+            'img': 'image',
+            'gt_bboxes': 'bboxes',
+        }),
+    dict(type='YOLOv5HSVRandomAug'),
+    dict(type='mmdet.RandomFlip', prob=0.5),
+    dict(
+        type='Polygon2Mask',
+        downsample_ratio=downsample_ratio,
+        mask_overlap=mask_overlap,
+        coco_style=False),
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_8xb16-300e_coco_instance.py:99]:
```suggestion
),
```

File: configs/yolov5/ins_seg/yolov5_ins_x-v61_syncbn_fast_8xb16-300e_coco_instance.py [hunk 16]
```diff
@@ -0,0 +1,15 @@
+_base_ = './yolov5_ins_m-v61_sync_bn_fast_8xb16-300e_coco_instance.py'  # noqa
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_x-v61_syncbn_fast_8xb16-300e_coco_instance.py:1]:
```suggestion
_base_ = './yolov5_ins_m-v61_syncbn_fast_8xb16-300e_coco_instance.py'  # noqa
```

File: configs/yolov5/ins_seg/yolov5_ins_l-v61_syncbn_fast_8xb16-300e_coco_instance.py [hunk 17]
```diff
@@ -0,0 +1,15 @@
+_base_ = './yolov5_ins_m-v61_sync_bn_fast_8xb16-300e_coco_instance.py'  # noqa
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_l-v61_syncbn_fast_8xb16-300e_coco_instance.py:1]:
```suggestion
_base_ = './yolov5_ins_m-v61_syncbn_fast_8xb16-300e_coco_instance.py'  # noqa
```

File: configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_non_overlap_8xb16-300e_coco_instance.py [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+            'img': 'image',
+            'gt_bboxes': 'bboxes',
+        }),
+    dict(type='YOLOv5HSVRandomAug'),
+    dict(type='mmdet.RandomFlip', prob=0.5),
+    dict(
+        type='Polygon2Mask',
+        downsample_ratio=_base_.downsample_ratio,
+        mask_overlap=mask_overlap,
+        coco_style=False),
```
Comment (hhaAndroid) [inline configs/yolov5/ins_seg/yolov5_ins_s-v61_syncbn_fast_non_overlap_8xb16-300e_coco_instance.py:43]:
```suggestion
),
```

**Comment by anying111** [comment]
This is a very good branch, can it be applied to yolov8?😊😊😊

**Comment by JosonChan1998** [comment]
> This is a very good branch, can it be applied to yolov8?😊😊😊

Yes, we will support yolov8-ins in the future.

**Review by hhaAndroid — APPROVED** [review]

**Comment by kateob** [comment]
Thanks for constructing this feature! I was able to train a yolov5 model for instance segmentation. The accuracy of the resulting pytorch model looked good. However, I was not able to convert this model to onnx with postprocessing. I tried instance-seg_onnxruntime_static.py in the mmdeploy instance segmentation configs but got the following error (apologies if this is too much information):


Traceback (most recent call last):
  File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py", line 107, in __call__
    ret = func(*args, **kwargs)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/pytorch2onnx.py", line 98, in torch2onnx
    export(
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py", line 356, in _wrap
    return self.call_function(func_name_, *args, **kwargs)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py", line 326, in call_function
    return self.call_function_local(func_name, *args, **kwargs)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py", line 275, in call_function_local
    return pipe_caller(*args, **kwargs)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py", line 107, in __call__
    ret = func(*args, **kwargs)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/onnx/export.py", line 131, in export
    torch.onnx.export(
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/__init__.py", line 350, in export
    return utils.export(
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py", line 163, in export
    _export(
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py", line 1074, in _export
    graph, params_dict, torch_out = _model_to_graph(
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/onnx/optimizer.py", line 27, in model_to_graph__custom_optimizer
    graph, params_dict, torch_out = ctx.origin_func(*args, **kwargs)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py", line 727, in _model_to_graph
    graph, params, torch_out, module = _create_jit_graph(model, args)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py", line 602, in _create_jit_graph
    graph, torch_out = _trace_and_get_graph_from_model(model, args)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py", line 517, in _trace_and_get_graph_from_model
    trace_graph, torch_out, inputs_states = torch.jit._get_trace_graph(
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/jit/_trace.py", line 1175, in _get_trace_graph
    outs = ONNXTracedModule(f, strict, _force_outplace, return_inputs, _return_inputs_states)(*args, **kwargs)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/jit/_trace.py", line 127, in forward
    graph, out = torch._C._create_graph_by_tracing(
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/jit/_trace.py", line 118, in wrapper
    outs.append(self.inner(*trace_inputs))
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1118, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/onnx/export.py", line 123, in wrapper
    return forward(*arg, **kwargs)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/codebase/mmdet/models/detectors/single_stage.py", line 89, in single_stage_detector__forward
    return __forward_impl(self, batch_inputs, data_samples=data_samples)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/core/optimizers/function_marker.py", line 266, in g
    rets = f(*args, **kwargs)
  File "/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/codebase/mmdet/models/detectors/single_stage.py", line 24, in __forward_impl
    output = self.bbox_head.predict(x, data_samples, rescale=False)
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/mmdet/models/dense_heads/base_dense_head.py", line 197, in predict
    predictions = self.predict_by_feat(
  File "/Workspace/Repos/MMdetection/mmyolo/mmyolo/models/dense_heads/yolov5_ins_head.py", line 681, in predict_by_feat
    results.masks = masks.bool()
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/mmengine/structures/instance_data.py", line 152, in __setattr__
    assert len(value) == len(self), 'The length of ' \
AssertionError: The length of values 1 is not consistent with the length of this :obj:`InstanceData` 18
07/25 18:24:22 - mmengine - ERROR -

**Comment by Caipengzhou** [comment]
> 感谢您构建此功能！我能够训练一个 yolov5 模型进行实例分割。生成的 pytorch 模型的准确性看起来不错。但是，我无法通过后处理将此模型转换为 onnx。我在 mmdeploy 实例分段配置中尝试了instance-seg_onnxruntime_static.py，但收到了以下错误（如果信息太多，请道歉）：
> 
> 回溯（最近调用最后一次）：文件“/usr/lib/python3.9/multiprocessing/process.py”，第 315 行，_bootstrap self.run（） 文件“/usr/lib/python3.9/multiprocessing/process.py”，第 108 行，运行self._target（*self._args， **self._kwargs） 文件“/workspace/repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py”，第 107 行**，调用** ret = func（*args， **kwargs）文件“/workspace/repos/MMdetection/mmdeploy/mmdeploy/apis/pytorch2onnx.py”，第 98 行，在 torch2onnx 导出中（文件“/workspace/repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py”，第 356 行，_换行返回 self.call_function（func_name_， *args， **kwargs）文件“/workspace/repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py”，第 326 行，call_function返回 self.call_function_local（func_name， *args， **kwargs） 文件 “/workspace/repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py”，第 275 行，call_function_local返回 pipe_caller（*args， **kwargs） 文件 “/workspace/repos/MMdetection/mmdeploy/mmdeploy/apis/core/pipeline_manager.py”，第 107 行**，调用** ret = func（*args， **kwargs）文件“/workspace/repos/MMdetection/mmdeploy/mmdeploy/apis/onnx/export.py”，第 131 行，在 export torch.onnx.export（ 文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/**init**.py”，第 350 行，在 export 中返回 utils.export（文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py”，第 163 行，在导出_export（文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py”，第 1074 行，_export图中，params_dict，torch_out = _model_to_graph（文件“/workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/onnx/optimizer.py”，第 27 行，model_to_graph__custom_optimizer图，params_dict，torch_out = ctx.origin_func（*args， **kwargs）文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py”，第 727 行，_model_to_graph图中，参数，torch_out，模块 = _create_jit_graph（model， args）文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py”，第 602 行，_create_jit_graph图中，torch_out = _trace_and_get_graph_from_model（model， args）文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/onnx/utils.py”，第 517 行，_trace_and_get_graph_from_modeltrace_graph、torch_out、inputs_states = torch.jit._get_trace_graph（ 文件 “/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/jit/_trace.py”，第 1175 行，_get_trace_graph outs = ONNXTracedModule（f， strict， _force_outplace， return_inputs， _return_inputs_states）（*args， **kwargs） 文件 “/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/nn/modules/module.py“，第 1130 行，在_call_impl返回 forward_call（*input， **kwargs） 文件”/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/jit/_trace.py“，第 127 行，在前向图中，输出 = torch._C._create_graph_by_tracing（文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/jit/_trace.py”，第 118 行，在包装器 outs.append（self.inner（*trace_inputs）） 文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/nn/modules/module.py”，第 1130 行，在_call_impl返回forward_call（*input， **kwargs）文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/torch/nn/modules/module.py”，第 1118 行，在_slow_forward结果 = self.forward（*input， **kwargs）文件“/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/apis/onnx/export.py”，第 123 行，在包装器中返回 forward（*arg， **kwargs）文件“/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/codebase/mmdet/models/detectors/single_stage.py”，第 89 行，返回single_stage_detector__forward __forward_impl（self， batch_inputs， data_samples=data_samples）文件“/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/core/optimizers/function_marker.py”，第 266 行，在 g rets = f（*args， **kwargs）文件“/Workspace/Repos/MMdetection/mmdeploy/mmdeploy/codebase/mmdet/models/detectors/single_stage.py”，第 24 行，在__forward_impl输出 = self.bbox_head.predict（x， data_samples， rescale=False） 文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/mmdet/models/dense_heads/base_dense_head.py”，第 197 行，在 predict predictions = self.predict_by_feat（文件“/Workspace/Repos/MMdetection/mmyolo/mmyolo/models/dense_heads/yolov5_ins_head.py”，第 681 行，predict_by_feat results.masks = masks.bool（） 文件“/local_disk0/.ephemeral_nfs/envs/pythonEnv-dab14687-d92c-483a-8722-5b2abe6392b0/lib/python3.9/site-packages/mmengine/structures/instance_data.py”，第 152 行，在 **setattr** assert len（value） == len（self）， '' 的长度AssertionError： 值 1 的长度与 this 的长度不一致 ：obj： 18 07/25 18：24：22 - mmengine - 错误 -`InstanceData`

Hello, may I ask if you have solved this problem? I also encountered this error when converting tensorrt. Looking forward to your reply.

**Pull request merged**